### DISCUSSION TITLE
How do I disable it for the react tags?

### DISCUSSION THREAD

**Discussion by tiago138** [HiPhish/rainbow-delimiters.nvim#29] [General]
How do I disable it for the react tags?

**Comment by HiPhish** [comment]
Do you want to disable it for React entirely, or only for React tags?

**Reply by tiago138** [reply]
Only for the tags.

**Reply by HiPhish** [reply]
You can change which query is used in the configuration. For Javascript and TSX files there is the `rainbow-parens` query which only covers parentheses without tags. See `:h rb-delimiters-setup` and `:h rb-delimiters-query`. You could also write your own custom query if none of the included ones are to your liking.

```vim
" Vim script
let g:rainbow_delimiters = { 'query': { 'tsx': 'rainbow-parens' } }
```

```lua
-- Lua
vim.g.rainbow_delimiters = { query = { tsx = 'rainbow-parens' } }
```

**Reply by tiago138** [reply]
Thanks!

**Reply by patelharsh9797** [reply]
is it same for jsx? I tried below ones but didn't work for jsx!!!

```
vim.g.rainbow_delimiters = { query = { tsx = "rainbow-parens", jsx = "rainbow-parens" } }
```

**Reply by HiPhish** [reply]
You are using the wrong language string. To get the right one evaluate in Lua `vim.treesitter.language.get_lang(<Vim filetype>)` where `<vim filetype>` is the file type assigned to the buffer by Neovim. In the case of Javascript and React that's `javascriptreact`. Putting these two together we get `vim.treesitter.language.get_lang('javascriptreact')`, which evaluates to `javascript`. Yup, regular Javascript and Javascript + React are the same language as far as Tree-sitter is concerned.

Why is there a difference between Vim file type and Tree-sitter language? Vim can first, so whatever Vim decided Neovim inherited. Tree-sitter came later, but it is used outside of Neovim as well, so it has its own names. In most cases the two are the same, e.g. Python is `python` for both. But there are exceptions, like how the Vim file type for Common Lisp is `lisp`, but the Tree-sitter language is `commonlisp`.
